build/python-client/Dockerfile

```diff
@@ -4,6 +4,19 @@ FROM --platform=x86_64 python:${PYTHON_VERSION}-buster
 
 RUN mkdir /proto
 
-RUN pip install grpcio grpcio-tools mypy-protobuf
+COPY client/python/pyproject.toml /code/pyproject.toml
+
+# Note that --use-feature=in-tree-build is needed until pip 21.3
+# (https://github.com/libAtoms/QUIP/issues/345)
+#
+# To make sure that the workdir is not changed (This might break something in the future),
+# We pip install from /code
+#
+# Then to install the `test` optional dependencies, we pip install from /code[test]
+# See https://stackoverflow.com/questions/46775346/what-do-square-brackets-mean-in-pip-install
```

Comment:
nit: SO links as doc links are weird. The explanation was probably OK without the link, I just generally prefer using primary sources. (Do not spend time fixing this though :D)

Reply:
I did this because I had never heard of this before, and I found barely any documentaition on it. Good to know tho :))